Modules/_decimal/_decimal.c

```diff
@@ -1603,8 +1604,9 @@ static PyObject *
 current_context(void)
 {
     PyThreadState *tstate = _PyThreadState_GET();
-    if (cached_context && cached_context->tstate == tstate) {
-        return (PyObject *)cached_context;
+    decimal_state *state = GLOBAL_STATE();
```

Comment:
`tstate` and `state` are too similar named variables. Please use `modstate` or another more distinct variable name.